paymentsheet/src/main/java/com/stripe/android/link/ui/wallet/WalletViewModel.kt

```diff
@@ -134,16 +139,35 @@ internal class WalletViewModel @Inject constructor(
 
         viewModelScope.launch {
             linkAccountManager.consumerState.filterNotNull().collectLatest { paymentDetailsState ->
-                if (paymentDetailsState.paymentDetails.isEmpty()) {
-                    navigateAndClearStack(LinkScreen.PaymentMethod)
+                val filteredPaymentDetails = paymentDetailsState.paymentDetails
+                    .filter { paymentMethodFilter?.invoke(it.details) != false }
+                    .toList()
+                if (filteredPaymentDetails.isEmpty()) {
+                    when (addPaymentMethodOptions.default) {
+                        AddPaymentMethodOption.Card -> {
+                            navigateAndClearStack(LinkScreen.PaymentMethod)
+                        }
+                        is AddPaymentMethodOption.Bank -> {
+                            navigateAndClearStack(LinkScreen.Wallet)
+                            presentAddBankAccount()
+                        }
```

Comment:
do we need to "navigate to wallet" if we're here already? or is this just to clear the stack?